hack/test-go.sh

```diff
@@ -256,18 +256,18 @@ runTests() {
   # must make sure the output from parallel runs is not mixed. To achieve this,
   # we spawn a subshell for each parallel process, redirecting the output to
   # separate files.
-  printf "%s\n" "${@}" | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
-    bash -c "set -o pipefail; _pkg=\"{}\"; _pkg_out=\${_pkg//\//_}; \
-        go test ${goflags[@]:+${goflags[@]}} \
-          ${KUBE_RACE} \
-          ${KUBE_TIMEOUT} \
-          -cover -covermode=\"${KUBE_COVERMODE}\" \
-          -coverprofile=\"${cover_report_dir}/\${_pkg}/${cover_profile}\" \
-          \"${KUBE_GO_PACKAGE}/\${_pkg}\" \
-          ${testargs[@]:+${testargs[@]}} \
-        | tee ${junit_filename_prefix:+\"${junit_filename_prefix}-\$_pkg_out.stdout\"} \
-        | grep \"${go_test_grep_pattern}\"" \
-      && test_result=$? || test_result=$?
+  if [ "$(uname)" == "Darwin" ]; then
+    args=(-I{} -n1)
+  else
+    args=(-I{} -n1 -d " ")
```

Comment:
The test result is, when I do not specific any delimiter, all paths are passed into run_go_test.sh at one time.

From: Jeff Grafton <notifications@github.com<mailto:notifications@github.com>>
Reply-To: kubernetes/kubernetes <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, May 10, 2016 at 9:39 AM
To: kubernetes/kubernetes <kubernetes@noreply.github.com<mailto:kubernetes@noreply.github.com>>
Cc: "Meng, Jesse" <fmeng@ebay.com<mailto:fmeng@ebay.com>>, Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [kubernetes/kubernetes] fix for test with coverage not working on osx (#24402)


In hack/test-go.sh<https://github.com/kubernetes/kubernetes/pull/24402#discussion_r62601120>:

> -  printf "%s\n" "${@}" | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
> -    bash -c "set -o pipefail; _pkg=\"{}\"; _pkg_out=\${_pkg//\//_}; \
> -        go test ${goflags[@]:+${goflags[@]}} \
> -          ${KUBE_RACE} \
> -          ${KUBE_TIMEOUT} \
> -          -cover -covermode=\"${KUBE_COVERMODE}\" \
> -          -coverprofile=\"${cover_report_dir}/\${_pkg}/${cover_profile}\" \
> -          \"${KUBE_GO_PACKAGE}/\${_pkg}\" \
> -          ${testargs[@]:+${testargs[@]}} \
> -        | tee ${junit_filename_prefix:+\"${junit_filename_prefix}-\$_pkg_out.stdout\"} \
> -        | grep \"${go_test_grep_pattern}\"" \
> -      && test_result=$? || test_result=$?
> +  if [ "$(uname)" == "Darwin" ]; then
> +    args=(-I{} -n1)
> +  else
> +    args=(-I{} -n1 -d " ")


is it necessary to set the delimiter here? I think GNU xargs uses spaces by default unless you pass -0.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/kubernetes/kubernetes/pull/24402/files/85d44468de7ecb0183d8fcc2a16b1bd09290edcd#r62601120>